situp-core/src/main/java/com/amazon/situp/parser/PipelineParser.java

```diff
@@ -1,32 +1,155 @@
+    /**
+     * Parses the configuration file into Pipeline
+     */
+    public Map<String, Pipeline> parseConfiguration() {
         try {
-            final PipelineConfiguration pipelineConfiguration = OBJECT_MAPPER.readValue(new File(configurationFileLocation),
-                    PipelineConfiguration.class);
-            PipelineConfigurationValidator.validate(pipelineConfiguration);
-            return pipelineConfiguration;
+            final Map<String, PipelineConfiguration> pipelineConfigurationMap = OBJECT_MAPPER.readValue(
+                    new File(configurationFileLocation), new TypeReference<Map<String, PipelineConfiguration>>() {
+                    });
+            final List<String> allPipelineNames = PipelineConfigurationValidator.
+                    validateAndGetPipelineNames(pipelineConfigurationMap);
+            final Map<String, Pipeline> pipelineMap = new HashMap<>();
+            for (String pipelineName : allPipelineNames) {
+                if (!pipelineMap.containsKey(pipelineName)) {
+                    buildPipelineFromConfiguration(pipelineName, pipelineConfigurationMap, pipelineMap);
+                }
+            }
+            return pipelineMap;
         } catch (IOException e) {
             throw new ParseException(format("Failed to parse the configuration file %s", configurationFileLocation), e);
         }
     }
-}
+
+    private void buildPipelineFromConfiguration(
+            final String pipelineName,
+            final Map<String, PipelineConfiguration> pipelineConfigurationMap,
+            final Map<String, Pipeline> pipelineMap) {
+        final PipelineConfiguration pipelineConfiguration = pipelineConfigurationMap.get(pipelineName);
+
+        final PluginSetting sourceSetting = pipelineConfiguration.getSource().getPluginSettings().get(0);
+        final Optional<Source> pipelineSource = getSourceIfPipelineType(pipelineName, sourceSetting,
+                pipelineMap, pipelineConfigurationMap);
+        final Source source = pipelineSource.orElseGet(() -> SourceFactory.newSource(sourceSetting));
+
+        final Buffer buffer = getBufferOrDefault(pipelineConfiguration.getBuffer());
+
+        final Configuration processorConfiguration = pipelineConfiguration.getProcessor();
+        final List<PluginSetting> processorPluginSettings = processorConfiguration.getPluginSettings();
+        final List<Processor> processors = processorPluginSettings.stream()
+                .map(ProcessorFactory::newProcessor)
+                .collect(Collectors.toList());
+        final int processorThreads = getWorkersOrDefault(processorConfiguration);
+        final int readBatchDelay = getDelayOrDefault(processorConfiguration);
+
+        final List<PluginSetting> sinkPluginSettings = pipelineConfiguration.getSink().getPluginSettings();
+        final List<Sink> sinks = sinkPluginSettings.stream().map(this::buildSinkOrConnector).collect(Collectors.toList());
```

Comment:
Just go ahead and make this one call:
```
final List<Sink> sinks = pipelineConfiguration.getSink().getPluginSettings().stream().map(...).collect(toList..);
```

Reply:
should it be `getSink()` or `getSinks()`?

Reply:
`getSink` because it is one Configuration object; which has `List<PluginSetting>` - will update the stream map as suggested.